packages/flutter_tools/lib/src/android/android_sdk.dart

```diff
@@ -430,23 +483,41 @@ class AndroidSdk {
     // See: http://stackoverflow.com/questions/14292698/how-do-i-check-if-the-java-jdk-is-installed-on-mac.
     if (platform.isMacOS) {
       try {
-        final String javaHomeOutput = globals.processUtils.runSync(
-          <String>['/usr/libexec/java_home', '-v', '1.8'],
-          throwOnError: true,
-          hideStdout: true,
-        ).stdout.trim();
+        // -v Filter versions (as if JAVA_VERSION had been set in the environment).
+        // It is unlikley that filtering to java version 1.8 is the right
+        // decsion here. That said, trying this on a mac shows the same jdk
```

Comment:
It looks like we could pass -V (instead of -v) to cause it to list all version (and parse from there) if we didn't want to rely on this, but FWIW I also get the same result no matter what version I pass to -v